packages/list/index.js

```diff
@@ -82,219 +85,188 @@ export default class List extends Component {
   }
 
   // Use onFocus as workaround because onFocusIn is not yet supported in React
   // https://github.com/facebook/react/issues/6410
-  onFocus = (e) => {
-    this.props.onFocus(e);
-    const index = this.getListItemIndexOfTarget_(e.target);
+  handleFocus = (e, index) => {
     this.foundation_.handleFocusIn(e, index);
   }
 
   // Use onBlur as workaround because onFocusOut is not yet supported in React
   // https://github.com/facebook/react/issues/6410
-  onBlur = (e) => {
-    this.props.onBlur(e);
-    const index = this.getListItemIndexOfTarget_(e.target);
+  handleBlur = (e, index) => {
     this.foundation_.handleFocusOut(e, index);
   }
 
-  render() {
+  renderListItem = (listItem, index) => {
     const {
-      /* eslint-disable no-unused-vars */
-      className,
       onKeyDown,
       onClick,
       onFocus,
       onBlur,
-      nonInterative,
-      dense,
-      avatarList,
-      twoLine,
-      singleSelection,
-      wrapFocus,
-      selectedIndex,
-      /* eslint-enable no-unused-vars */
-      children,
       ...otherProps
-    } = this.props;
-
-    return (
-      <ul
-        className={this.classes}
-        onKeyDown={this.onKeyDown}
-        onClick={this.onClick}
-        onFocus={this.onFocus}
-        onBlur={this.onBlur}
-        {...otherProps}
-      >
-        {React.Children.map(children, this.renderListItem)}
-      </ul>
-    );
-  }
+    } = listItem.props;
 
-  renderListItem = (listItem, index) => {
     const {
-      id,
-      /* eslint-disable no-unused-vars */
-      className,
-      childrenTabIndex,
-      /* eslint-enable no-unused-vars */
-      children,
-      ...otherProps
-    } = listItem.props;
+      focusListItemAtIndex,
+      followHrefAtIndex,
+      toggleCheckboxAtIndex,
+      listItemAttributes,
+      listItemClassNames,
+      listItemChildrenTabIndex,
+    } = this.state;
 
-    const idOrIndex = id || index.toString();
-    const props = Object.assign({
-      id: id,
-      className: this.getListItemClasses_(idOrIndex),
-      childrenTabIndex: this.state.listItemChildrenTabIndex[idOrIndex],
-      init: () => this.initListItem(idOrIndex),
-      ref: (listItem) => {
-        this.listItems_[idOrIndex] = listItem;
-        this.listItemIndices_[idOrIndex] = index;
-      },
+    const props = {
       ...otherProps,
-    },
-    this.state.listItemAttributes[idOrIndex]);
+      onKeyDown: (e) => {
+        onKeyDown();
+        this.handleKeyDown(e, index);
+      },
+      onClick: (e) => {
+        onClick();
+        this.handleClick(e, index);
+      },
+      onFocus: (e) => {
+        onFocus();
+        this.handleFocus(e, index);
+      },
+      onBlur: (e) => {
+        onBlur();
+        this.handleBlur(e, index);
+      },
+      shouldFocus: focusListItemAtIndex === index,
+      shouldFollowHref: followHrefAtIndex === index,
+      shouldToggleCheckbox: toggleCheckboxAtIndex === index,
+      attributesToSet: listItemAttributes[index],
```

Comment:
How about renaming `attributesToSet` and `classNamesToAdd` to `attributesFromList` and `classNamesFromList`? I think it is inferred that we want to add them.

Reply:
omg yes, I struggled hard with this naming 😅